src/setup_payload/Base38.h

```diff
@@ -17,20 +17,20 @@
 
 /**
  *    @file
- *      Utility header to encode an input into a Base38 String
+ *      Shared header for encoding and decoding Base38
  */
 
 #pragma once
 
 #include <core/CHIPError.h>
 
 #include <stdint.h>
-#include <string>
-#include <vector>
 
 namespace chip {
-// returns CHIP_NO_ERROR on successful decode
-CHIP_ERROR base38Decode(std::string base38, std::vector<uint8_t> & out);
-std::string base38Encode(const uint8_t * buf, size_t buf_len);
+
+static const char kCodes[] = { '0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I',
+                               'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', '-', '.' };
+static const uint8_t kBase38CharactersNeededInNBytesChunk[] = { 2, 4, 5 };
+static const uint8_t kRadix                                 = sizeof(kCodes) / sizeof(kCodes[0]);
```

Comment:
See `ArraySize` common utility in `src/lib/support/CodeUtils.h`  to replace this default `sizeof` division